src/Tools/dotnet-monitor/Auth/MonitorApiKeyConfiguration.cs

```diff
@@ -3,17 +3,22 @@
 // See the LICENSE file in the project root for more information.
 
 using Microsoft.AspNetCore.Authentication;
+using Microsoft.IdentityModel.Tokens;
+using System;
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
 
 namespace Microsoft.Diagnostics.Tools.Monitor
 {
-    internal sealed class ApiKeyAuthenticationOptions : AuthenticationSchemeOptions
+    /// <summary>
+    /// Represents internal version of <see cref="MonitorApiKeyOptions"/>.
+    /// This object contains the validation state of the object and the decoded
+    /// Json Web Key.
+    /// </summary>
+    internal class MonitorApiKeyConfiguration : AuthenticationSchemeOptions
```

Comment:
This now feels like it it an intermediate options class that could just be folded into JwtBearerOptions (not literally put the properties on there, but in its IPostConfigureOptions and IValidateOptions implementations, you can do the same logic and then remove this options class). This would reduce the complexity of this code a tiny bit and probably remove one or two of the extra reloads.

Reply:
Maybe don't do this simplification now. We can try it after the rest of it gets in.